app/models/metadata.rb

```diff
@@ -64,7 +68,14 @@ def self.sanitize_params(filters)
         query[filter['name']] = filter['options']
       end
     end
-    query.delete_if { |_k, v| v.empty? }
+    query.delete_if { |k, v| v.empty? || 'factsheet'.include?(k) }
+  end
+
+  def self.factsheet_filter(filters, data)
+    return data unless filters.each { |filter| 'factsheet'.include? filter['name'] }
+    sheet_filter = filters.select { |filter| filter['name'] == 'factsheet' }.first
+    return data if sheet_filter['options'].empty?
+    sheet_filter['options'].first ? data.where.not(factsheet: nil) : data.where(factsheet: nil)
```

Comment:
This looks fine overall, but I wonder if we should already consider the situation where a factsheet is an empty string rather then `nil`; I think it could happen

Reply:
In the dummy csv there is no evidence of this though. When the factsheet attribute is not present there is blank space

Reply:
That's fine then :)